oak/server/dev/oak_manager.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oak_manager.h"
+
+#include "absl/memory/memory.h"
+#include "asylo/grpc/auth/enclave_channel_credentials.h"
+#include "asylo/grpc/auth/enclave_server_credentials.h"
+#include "asylo/grpc/auth/null_credentials_options.h"
+#include "asylo/identity/descriptions.h"
+#include "asylo/identity/init.h"
+#include "asylo/util/logging.h"
+#include "include/grpcpp/grpcpp.h"
+
+namespace oak {
+
+OakManager::OakManager(absl::string_view _path)
+    : Service(), application_id_(0), runtime_(absl::make_unique<OakRuntime>()) {
+  LOG(INFO) << "Creating OakManager";
+}
+
+grpc::Status OakManager::CreateApplication(grpc::ServerContext* context,
+                                           const oak::CreateApplicationRequest* request,
+                                           oak::CreateApplicationResponse* response) {
+  std::string application_id = NewApplicationId();
```

Comment:
Should this move inside `OakRuntime`, and then expose a `GetApplicationId` method from it?

Reply:
I like it like this for now, as it allows each implementation to use a different way to create app ids